### DISCUSSION TITLE
How to use `--changed` before the initial commit?

### DISCUSSION THREAD

**Discussion by TomerAberbach** [vitest-dev/vitest#6319] [Q&A] [answered]
I have a `package.json` script that runs `vitest --watch=false --passWithNoTests --changed HEAD~1`, and it always works _except_ before committing the initial commit, because `HEAD~1` doesn't exist in that case. Wondering if anyone else has dealt with this and solved it?

**Comment by AriPerkkio** [comment, marked as answer]
Isn't that what the default value does?

> If no value is provided, it will run tests against uncommitted changes (including staged and unstaged).
> https://vitest.dev/guide/cli.html#changed

```diff
- --changed HEAD~1
+ --changed
```

**Reply by TomerAberbach** [reply]
Welp, you are correct! I don't know how I reached the conclusion that an argument was required. Thanks :)
